lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/driver/RunningWorkers.java

```diff
@@ -143,6 +156,29 @@ void launchTasklet(final Tasklet tasklet) {
 
   /**
    * Concurrency: Called by multiple threads.
+   * Parameter: Same taskletId can come in multiple times.
+   */
+  void cancelTasklet(final boolean mayInterruptIfRunning, final int taskletId) {
+    lock.lock();
+    try {
+      final String workerId = getWhereTaskletWasScheduledTo(taskletId); // This is not ideal.
+      if (workerId == null) {
+        // launchTasklet called but not yet running.
+        cancelledTasklets.add(taskletId);
+        return;
+      }
+
+      if (mayInterruptIfRunning) {
+        LOG.log(Level.WARNING, "Cancelling running Tasklet with ID " + taskletId + ".");
```

Comment:
Should this be `WARNING`? I think the user requested to cancel the Tasklet, so wouldn't it be a desired result?